### DISCUSSION TITLE
Some error in my whisper Project in ubuntu

### DISCUSSION THREAD

**Discussion by ahmednmwothoq** [openai/whisper#2264] [Show and tell]
i have error in my project , please can you help me to solve it !!!
i create environment for project and run Gunicorn with nginx 

#### **The Errors**
1-
Jul 11 00:07:47   gunicorn[74177]:     return forward_call(*args, **kwargs)
Jul 11 00:07:47   gunicorn[74177]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jul 11 00:07:47   gunicorn[74177]:   File "/home/aiw/env/whisper/lib/python3.12/site-packages/whisper/model.py", line 37, in forward
Jul 11 00:07:47  gunicorn[74177]:     return F.linear(
Jul 11 00:07:47  gunicorn[74177]:            ^^^^^^^^^
Jul 11 00:07:47  gunicorn[74177]:   File "/home/aiw/env/whisper/lib/python3.12/site-packages/gunicorn/workers/base.py", line 203, in handle_abort
Jul 11 00:07:47  gunicorn[74177]:     sys.exit(1)
Jul 11 00:07:47  gunicorn[74177]: SystemExit: 1

2-
Jul 11 09:14:01  gunicorn[78237]:     wv, qk = self.qkv_attention(q, k, v, mask)
Jul 11 09:14:01  gunicorn[78237]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jul 11 09:14:01  gunicorn[78237]:   File "/home/aiw/env/whisper/lib/python3.12/site-packages/whisper/model.py", line 108, in qkv_attention
Jul 11 09:14:01  gunicorn[78237]:     return (w @ v).permute(0, 2, 1, 3).flatten(start_dim=2), qk.detach()
Jul 11 09:14:01  gunicorn[78237]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jul 11 09:14:01  gunicorn[78237]:   File "/home/aiw/env/whisper/lib/python3.12/site-packages/gunicorn/workers/base.py", line 203, in handle_abort
Jul 11 09:14:01  gunicorn[78237]:     sys.exit(1)
Jul 11 09:14:01  gunicorn[78237]: SystemExit: 1

3-
Jul 11 10:29:52  gunicorn[79907]:     wv, qk = self.qkv_attention(q, k, v, mask)
Jul 11 10:29:52  gunicorn[79907]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jul 11 10:29:52  gunicorn[79907]:   File "/home/aiw/env/whisper/lib/python3.12/site-packages/whisper/model.py", line 105, in qkv_attention
Jul 11 10:29:52  gunicorn[79907]:     qk = qk.float()
Jul 11 10:29:52  gunicorn[79907]:          ^^^^^^^^^^
Jul 11 10:29:52  gunicorn[79907]:   File "/home/aiw/env/whisper/lib/python3.12/site-packages/gunicorn/workers/base.py", line 203, in handle_abort
Jul 11 10:29:52  gunicorn[79907]:     sys.exit(1)
Jul 11 10:29:52  gunicorn[79907]: SystemExit: 1

### **My Code**

```
def extract_features(audio_path, start, end):
    """ Extracting audio features using librosa """
    y, sr = librosa.load(audio_path, sr=None, offset=start, duration=end - start)
    mfcc = librosa.feature.mfcc(y=y, sr=sr, n_mfcc=13)
    mfcc_delta = librosa.feature.delta(mfcc)
    mfcc_delta2 = librosa.feature.delta(mfcc, order=2)
    features = np.concatenate((np.mean(mfcc.T, axis=0), np.mean(mfcc_delta.T, axis=0), np.mean(mfcc_delta2.T, axis=0)))
    return features


def cluster_speakers(features, n_clusters=2):
    """ Classification of speakers using KMeans """
    kmeans = KMeans(n_clusters=n_clusters)
    kmeans.fit(features)
    return kmeans.labels_


def transcribe_audio(audio_path, language):
    # Convert audio to suitable format using FFmpeg
    wav_path = uuid.uuid4().hex[:6].upper() + ".wav"
    convert_audio(audio_path, wav_path)
    # Use Whisper AI to convert audio to text
    transcription = whisper_model.transcribe(wav_path, language=language)
    # Extract audio features for each piece of text
    segments = transcription['segments']
    features = []
    for segment in segments:
        start = segment['start']
        end = segment['end']
        feature = extract_features(wav_path, start, end)
        features.append(feature)

    # Convert features to numpy array
    features = np.array(features)

    # Classification of speakers using acoustic features
    labels = cluster_speakers(features, n_clusters=2)  # يمكنك تعديل عدد المجموعات حسب الحاجة

    # Combine selection and transformation results
    result = []
    for segment, label in zip(segments, labels):
        text = segment['text']
        result.append({
            'speaker': f"Speaker {label + 1}",
            'start': segment['start'],
            'end': segment['end'],
            'text': text
        })

    # Delete the temporary file
    os.remove(wav_path)

    return result
```

**Comment by ahmednmwothoq** [comment]
but in local , the code work well

**Reply by glangford** [reply]
Could the problem be with Python 3.12

**Reply by ahmednmwothoq** [reply]
python is work good , whisper too 
i test it , by using command "whisper audio.mp3 --model medium"
